### PR TITLE
docs(storybook): inferred targets

### PR DESCRIPTION
**Pull request by mandarini**
Inferred targets docs for storybook

### CODE REVIEW

**Review by isaacplmann — CHANGES_REQUESTED** [review]
Just need to note whether outputs are static or read from a config file

File: docs/shared/recipes/storybook/storybook-inferred-targets.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+| --------- | --------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
+| `command` | `storybook build`                                                     | Build Storybook for the project                                                                   |
+| `cache`   | `true`                                                                | Automatically cache the task                                                                      |
+| `inputs`  | `production`                                                          | Break the cache if non-test files in the project change                                           |
+| `inputs`  | `{ externalDependencies: [ "storybook", "@storybook/test-runner" ] }` | Break the cache if the version of the `storybook` package or the `@storybook/test-runner` changes |
+| `outputs` | `["{workspaceRoot}/dist/storybook/{projectRoot}"]`                    | The output directory of the build                                                                 |
```
[the remark this answers is not in the document]
Comment (mandarini) [reply docs/shared/recipes/storybook/storybook-inferred-targets.md:61]: It's static unfortunately, until we manage to read args from CLI

File: docs/shared/recipes/storybook/storybook-inferred-targets.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+| `cache`    | `true`                                                                                      | Automatically cache the task                                                                                                                   |
+| `inputs`   | `production`                                                                                | Break the cache if non-test files in the project change                                                                                        |
+| `inputs`   | `{ externalDependencies: [ "storybook", "@storybook/angular", "@storybook/test-runner" ] }` | Break the cache if the version of the `storybook` package, or the `@storybook/angular` package or the `@storybook/test-runner` package changes |
+| `outputs`  | `["{workspaceRoot}/dist/storybook/{projectRoot}"]`                                          | The output directory of the build                                                                                                              |
```
[the remark this answers is not in the document]
Comment (mandarini) [reply docs/shared/recipes/storybook/storybook-inferred-targets.md:86]: It's static unfortunately, until we manage to read args from CLI

File: docs/shared/recipes/storybook/storybook-inferred-targets.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+      "cache": true,
+      "inputs": [
+        "production",
+        "^production",
+        { "externalDependencies": ["storybook", "@storybook/test-runner"] }
+      ],
+      "outputs": ["{workspaceRoot}/dist/storybook/{projectRoot}"],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "storybook build --config-dir my-project/.storybook --output-dir <full path to workspace root>/dist/storybook/my-project"
```
[the remark this answers is not in the document]
Comment (mandarini) [reply docs/shared/recipes/storybook/storybook-inferred-targets.md:120]: ohhhhhhh interesting!!!!!!!!!! thanks, let me see if it works well without the full path

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**